hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/BackendType.java

```diff
@@ -1,5 +1,5 @@
 package org.hypertrace.traceenricher.enrichment.enrichers;
 
 public enum BackendType {
-  UNKNOWN, HTTP, HTTPS, GRPC, REDIS, MONGO, JDBC, RABBIT_MQ
+  UNKNOWN, HTTP, HTTPS, GRPC, REDIS, MONGO, JDBC, RABBIT_MQ, KAFKA, AWS_SQS
```

Comment:
I think we need to update the enrichment of the protocol field in - https://github.com/hypertrace/hypertrace-trace-enricher/blob/91fe3972226453151c7fc3bdcb8aadb7a506f295/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/SpanTypeAttributeEnricher.java#L64

@rish691, is it right?

Reply:
This is what should be extended in the setting protocol - https://github.com/open-telemetry/opentelemetry-specification/blob/main/semantic_conventions/trace/messaging.yaml#L38, however, we need to see what do we get as URL if the protocol is kafka

Reply:
I think we should also extend `rabbitmq` with `MESSAGING_URL` - `messaging.url` attribute of otel.

Reply:
> I think we need to update the enrichment of the protocol field in - https://github.com/hypertrace/hypertrace-trace-enricher/blob/91fe3972226453151c7fc3bdcb8aadb7a506f295/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/SpanTypeAttributeEnricher.java#L64
@rish691, is it right?

Yes, this is needed to correctly populate the protocol for the span. 

>This is what should be extended in the setting protocol - https://github.com/open-telemetry/opentelemetry-specification/blob/main/semantic_conventions/trace/messaging.yaml#L38, however, we need to see what do we get as URL if the protocol is kafka, sqs, etc

I think this is transport protocol, slightly different from our convention of protocol (which is currently 1-1 mapping with backend type).

Reply:
For rabbitmq, otel semantics are already being used to get routing key, we just need an additional check for `messaging.system=rabbitmq` I guess.